### PR TITLE
rpcv05 Implement GetTransactionStatus

### LINKED ISSUE #411: Addition of `starknet_getTransactionStatus` method and TXN_STATUS type

### PR DESCRIPTION
**Pull request by rianhughes**
closes https://github.com/NethermindEth/starknet.go/issues/411

### CODE REVIEW

File: rpc/transaction_test.go [hunk 1]
```diff
@@ -309,3 +309,30 @@ func TestDeployOrDeclareReceipt(t *testing.T) {
...
+	testConfig := beforeEach(t)
+
+	type testSetType struct {
+		TxnHash      *felt.Felt
+		ExpextedResp GetTxnStatusResp
+	}
+
+	testSet := map[string][]testSetType{
+		"mock": {},
+		"testnet": {
```
Comment (joshklop) [inline rpc/transaction_test.go:324]: This test does not run by default (only `"mock"` tests are executed, it seems). Are all "test environments" executed by the CI, or are only mock tests run?

Comment (joshklop) [reply rpc/transaction_test.go:324]: Ah I see it appears those tests are run by the CI :+1:

File: rpc/types_transaction_receipt.go [hunk 2]
```diff
@@ -264,3 +264,18 @@ func unmarshalTransactionReceipt(t interface{}) (TransactionReceipt, error) {
...
+type TxnStatus string
+
+const (
+	TxnStatus_Recieved       TxnStatus = "RECEIVED"
+	TxnStatus_Rejected       TxnStatus = "REJECTED"
+	TxnStatus_Accepted_On_L2 TxnStatus = "ACCEPTED_ON_L2"
+	TxnStatus_Accepted_On_L1 TxnStatus = "ACCEPTED_ON_L1"
+)
+
+type GetTxnStatusResp struct {
```
Comment (cicr99) [inline rpc/types_transaction_receipt.go]: Could we rename this to just `TxnStatusResp`. I would rather use the verb at the beginning ("Get" in this case) when it comes to a method so you can emphasize the action, on a type however, I'd rather not have it

File: rpc/transaction.go [hunk 3]
```diff
@@ -106,3 +106,16 @@ func (provider *Provider) TransactionReceipt(ctx context.Context, transactionHas
 	}
 	return receipt.TransactionReceipt, nil
 }
+
+// GetTransactionStatus gets the transaction status (possibly reflecting that the tx is still in the mempool, or dropped from it)
+func (provider *Provider) GetTransactionStatus(ctx context.Context, transactionHash *felt.Felt) (*GetTxnStatusResp, error) {
```
Comment (cicr99) [inline rpc/transaction.go]: Shoudln't this method be added to `RpcProvider` interface?

Comment (cicr99) [reply rpc/transaction.go]: Let's also add it in the Rpc supported features board in the readme

Comment (cicr99) [reply rpc/transaction.go]: Can the comment be updated to have the same format as #415 ?

File: rpc/transaction.go [hunk 4]
```diff
@@ -106,3 +106,16 @@ func (provider *Provider) TransactionReceipt(ctx context.Context, transactionHas
...
+	var receipt GetTxnStatusResp
+	err := do(ctx, provider.c, "starknet_getTransactionStatus", &receipt, transactionHash)
+	if err != nil {
+		if errors.Is(err, ErrHashNotFound) {
```
Comment (cicr99) [inline rpc/transaction.go]: I think we should use `tryUnwrapToRPCErr` to process the error instead

File: rpc/types_transaction_receipt.go [hunk 5]
```diff
@@ -264,3 +264,18 @@ func unmarshalTransactionReceipt(t interface{}) (TransactionReceipt, error) {
 
 	return nil, fmt.Errorf("unknown transaction type: %v", t)
 }
+
+// The finality status of the transaction, including the case the txn is still in the mempool or failed validation during the block construction phase
+type TxnStatus string
+
+const (
+	TxnStatus_Recieved       TxnStatus = "RECEIVED"
```
Comment (cicr99) [inline rpc/types_transaction_receipt.go]: Typo TxnStatus_Recieved -> TxnStatus_Received

File: rpc/transaction_test.go [hunk 6]
```diff
@@ -309,3 +309,30 @@ func TestDeployOrDeclareReceipt(t *testing.T) {
...
 	}
 }
+
+// TestGetTransactionStatus tests starknet_getTransactionStatus
+func TestGetTransactionStatus(t *testing.T) {
+	testConfig := beforeEach(t)
+
+	type testSetType struct {
+		TxnHash      *felt.Felt
+		ExpextedResp GetTxnStatusResp
```
Comment (cicr99) [inline rpc/transaction_test.go]: Typo ExpextedResp -> ExpectedResp

File: examples/deployAccount/account
Comment (cicr99) [inline examples/deployAccount/account:1]: Why was this binary added?

File: examples/simpleCall/account
Comment (cicr99) [inline examples/simpleCall/account:1]: Same question about this one

File: examples/deployAccount/account
Comment (rianhughes) [reply examples/deployAccount/account:1]: Must have been added accidentally after building the examples

File: examples/simpleCall/account
Comment (rianhughes) [reply examples/simpleCall/account:1]: Must have been added accidentally after building the examples

**Review by cicr99 — APPROVED** [review]

**Pull request merged**